src/lightning/fabric/strategies/fsdp.py

```diff
@@ -270,10 +271,13 @@ def module_to_device(self, module: Module) -> None:
         pass
 
     @contextmanager
-    def module_init_context(self) -> Generator[None, None, None]:
+    def module_init_context(self, empty_init: Optional[bool] = None) -> Generator[None, None, None]:
         # TODO: Use the meta device and reset parameters after https://github.com/pytorch/pytorch/issues/90465
         # is resolved. For now, the module will get moved to the device in `setup_module`.
-        with self.precision.init_context(), self.module_sharded_context():
+        empty_init_context = (
+            _EmptyInit(enabled=(empty_init is not False)) if _TORCH_GREATER_EQUAL_1_13 else nullcontext()
```

Comment:
@awaelchli We might want to reconsider doing `empty_init` by default on FSDP. I just went down a rabbit hole chasing NaNs on FSDP when all I needed to change was `empty_init=False`. Since the default was to skip initialization and I wasn't loading a full checkpoint, the weights were initialized to garbage.

Since there's no way for us to know if the user will load a full checkpoint after initialization, I find it safer to not do this by default.

context: https://github.com/Lightning-AI/lit-gpt/pull/193

Reply:
That wasn't the intention, sorry. The plan was to do this automatically if we have fake tensor / support for materialization (torchdistx-style). In the meantime, it would be safer to do:

```suggestion
            _EmptyInit(enabled=bool(empty_init) if _TORCH_GREATER_EQUAL_1_13 else nullcontext()
```

Reply:
The comment above there

>         # TODO: Use the meta device and reset parameters after https://github.com/pytorch/pytorch/issues/90465

is what I ultimately wanted to achieve

Reply:
Same. While it's a cool feature, having this default to true is very disruptive and hard to debug.